// ng-semantic/bundles/ng-semantic.js
System.registerDynamic("ng-semantic/menu/menu", ["@angular/core"], true, function($__require, exports, module) {
  "use strict";
  ;
  var define,
      global = this,
      GLOBAL = this;
  var __decorate = (this && this.__decorate) || function(decorators, target, key, desc) {
    var c = arguments.length,
        r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
        d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
      r = Reflect.decorate(decorators, target, key, desc);
    else
      for (var i = decorators.length - 1; i >= 0; i--)
        if (d = decorators[i])
          r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
  };
  var __metadata = (this && this.__metadata) || function(k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
      return Reflect.metadata(k, v);
  };
  var core_1 = $__require('@angular/core');
  var SemanticMenuComponent = (function() {
    function SemanticMenuComponent() {}
    __decorate([core_1.Input("logo"), __metadata('design:type', String)], SemanticMenuComponent.prototype, "logo", void 0);
    SemanticMenuComponent = __decorate([core_1.Component({
      changeDetection: core_1.ChangeDetectionStrategy.OnPush,
      selector: "sm-menu",
      template: "\n<a href=\"#/\" *ngIf=\"logo\" class=\"header item\">\n    <img class=\"logo\" alt=\"{{title}}\" src=\"{{logo}}\">\n</a>\n\n<ng-content></ng-content>\n"
    }), __metadata('design:paramtypes', [])], SemanticMenuComponent);
    return SemanticMenuComponent;
  }());
  exports.SemanticMenuComponent = SemanticMenuComponent;
  return module.exports;
});

System.registerDynamic("ng-semantic/message/message", ["@angular/core"], true, function($__require, exports, module) {
  "use strict";
  ;
  var define,
      global = this,
      GLOBAL = this;
  var __decorate = (this && this.__decorate) || function(decorators, target, key, desc) {
    var c = arguments.length,
        r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
        d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
      r = Reflect.decorate(decorators, target, key, desc);
    else
      for (var i = decorators.length - 1; i >= 0; i--)
        if (d = decorators[i])
          r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
  };
  var __metadata = (this && this.__metadata) || function(k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
      return Reflect.metadata(k, v);
  };
  var core_1 = $__require('@angular/core');
  var SemanticMessageComponent = (function() {
    function SemanticMessageComponent() {}
    __decorate([core_1.Input("icon"), __metadata('design:type', String)], SemanticMessageComponent.prototype, "icon", void 0);
    SemanticMessageComponent = __decorate([core_1.Component({
      changeDetection: core_1.ChangeDetectionStrategy.OnPush,
      encapsulation: core_1.ViewEncapsulation.None,
      selector: "sm-message",
      styles: ["sm-message { display: block; }"],
      template: "<i *ngIf=\"icon\" class=\"icon {{icon}}\"></i>\n  <p><ng-content></ng-content></p>"
    }), __metadata('design:paramtypes', [])], SemanticMessageComponent);
    return SemanticMessageComponent;
  }());
  exports.SemanticMessageComponent = SemanticMessageComponent;
  return module.exports;
});

System.registerDynamic("ng-semantic/popup/tooltip", ["@angular/core"], true, function($__require, exports, module) {
  "use strict";
  ;
  var define,
      global = this,
      GLOBAL = this;
  var __decorate = (this && this.__decorate) || function(decorators, target, key, desc) {
    var c = arguments.length,
        r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
        d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
      r = Reflect.decorate(decorators, target, key, desc);
    else
      for (var i = decorators.length - 1; i >= 0; i--)
        if (d = decorators[i])
          r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
  };
  var __metadata = (this && this.__metadata) || function(k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
      return Reflect.metadata(k, v);
  };
  var core_1 = $__require('@angular/core');
  var SMTooltipDirective = (function() {
    function SMTooltipDirective(element) {
      this.element = element;
    }
    SMTooltipDirective.prototype.ngOnInit = function() {
      this.element.element.nativeElement.setAttribute("data-position", this.smDirPosition || "top center");
      this.element.element.nativeElement.setAttribute("data-tooltip", this.smDirTooltip);
    };
    __decorate([core_1.Input(), __metadata('design:type', String)], SMTooltipDirective.prototype, "smDirTooltip", void 0);
    __decorate([core_1.Input(), __metadata('design:type', String)], SMTooltipDirective.prototype, "smDirPosition", void 0);
    SMTooltipDirective = __decorate([core_1.Directive({selector: "[smDirTooltip]"}), __metadata('design:paramtypes', [core_1.ViewContainerRef])], SMTooltipDirective);
    return SMTooltipDirective;
  }());
  exports.SMTooltipDirective = SMTooltipDirective;
  return module.exports;
});

System.registerDynamic("ng-semantic/popup/popup", ["@angular/core"], true, function($__require, exports, module) {
  "use strict";
  ;
  var define,
      global = this,
      GLOBAL = this;
  var __decorate = (this && this.__decorate) || function(decorators, target, key, desc) {
    var c = arguments.length,
        r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
        d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
      r = Reflect.decorate(decorators, target, key, desc);
    else
      for (var i = decorators.length - 1; i >= 0; i--)
        if (d = decorators[i])
          r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
  };
  var __metadata = (this && this.__metadata) || function(k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
      return Reflect.metadata(k, v);
  };
  var core_1 = $__require('@angular/core');
  var SemanticPopupComponent = (function() {
    function SemanticPopupComponent() {
      this.visible = false;
    }
    SemanticPopupComponent.prototype.show = function(element, data) {
      if (data === void 0) {
        data = {};
      }
      if (!this.visible) {
        this.visible = true;
        this.element = element.srcElement;
        jQuery(element.srcElement).popup({
          closable: true,
          exclusive: true,
          lastResort: true,
          on: "click",
          popup: this.popup.nativeElement,
          position: (data.hasOwnProperty("position")) ? data.position : "bottom center",
          preserve: true
        }).popup("toggle");
      }
    };
    SemanticPopupComponent.prototype.hide = function() {
      if (this.visible && this.element) {
        this.visible = false;
        jQuery(this.element).popup("hide");
      }
    };
    __decorate([core_1.ViewChild("popup"), __metadata('design:type', core_1.ElementRef)], SemanticPopupComponent.prototype, "popup", void 0);
    SemanticPopupComponent = __decorate([core_1.Component({
      selector: "sm-popup",
      template: "<div class=\"ui popup very wide {{selector}}\" #popup>\n    <div class=\"content\">\n        <ng-content></ng-content>\n    </div>\n</div>"
    }), __metadata('design:paramtypes', [])], SemanticPopupComponent);
    return SemanticPopupComponent;
  }());
  exports.SemanticPopupComponent = SemanticPopupComponent;
  return module.exports;
});

System.registerDynamic("ng-semantic/modal/modal", ["@angular/core"], true, function($__require, exports, module) {
  "use strict";
  ;
  var define,
      global = this,
      GLOBAL = this;
  var __decorate = (this && this.__decorate) || function(decorators, target, key, desc) {
    var c = arguments.length,
        r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
        d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
      r = Reflect.decorate(decorators, target, key, desc);
    else
      for (var i = decorators.length - 1; i >= 0; i--)
        if (d = decorators[i])
          r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
  };
  var __metadata = (this && this.__metadata) || function(k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
      return Reflect.metadata(k, v);
  };
  var core_1 = $__require('@angular/core');
  var SemanticModalComponent = (function() {
    function SemanticModalComponent() {}
    SemanticModalComponent.prototype.show = function(data) {
      jQuery(this.modal.nativeElement).modal(data || {}).modal("toggle");
    };
    SemanticModalComponent.prototype.hide = function() {
      jQuery(this.modal.nativeElement).modal("hide");
    };
    __decorate([core_1.Input(), __metadata('design:type', String)], SemanticModalComponent.prototype, "class", void 0);
    __decorate([core_1.Input(), __metadata('design:type', String)], SemanticModalComponent.prototype, "title", void 0);
    __decorate([core_1.Input(), __metadata('design:type', String)], SemanticModalComponent.prototype, "icon", void 0);
    __decorate([core_1.ViewChild("modal"), __metadata('design:type', core_1.ElementRef)], SemanticModalComponent.prototype, "modal", void 0);
    SemanticModalComponent = __decorate([core_1.Component({
      changeDetection: core_1.ChangeDetectionStrategy.OnPush,
      selector: "sm-modal",
      template: "<div class=\"ui modal {{class}}\" #modal>\n    <i class=\"close icon\"></i>\n    <div [ngClass]=\"{'icon': icon}\" class=\"ui header\">\n        <i *ngIf=\"icon\" class=\"icon {{icon}}\"></i>\n        {{title}}\n    </div>\n    <div class=\"content\">\n        <ng-content select=\"modal-content\"></ng-content>\n    </div>\n    <div class=\"actions\">\n        <ng-content select=\"modal-actions\"></ng-content>\n    </div>\n</div>"
    }), __metadata('design:paramtypes', [])], SemanticModalComponent);
    return SemanticModalComponent;
  }());
  exports.SemanticModalComponent = SemanticModalComponent;
  return module.exports;
});

System.registerDynamic("ng-semantic/segment/segment", ["@angular/core"], true, function($__require, exports, module) {
  "use strict";
  ;
  var define,
      global = this,
      GLOBAL = this;
  var __decorate = (this && this.__decorate) || function(decorators, target, key, desc) {
    var c = arguments.length,
        r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
        d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
      r = Reflect.decorate(decorators, target, key, desc);
    else
      for (var i = decorators.length - 1; i >= 0; i--)
        if (d = decorators[i])
          r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
  };
  var __metadata = (this && this.__metadata) || function(k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
      return Reflect.metadata(k, v);
  };
  var core_1 = $__require('@angular/core');
  var SemanticSegmentComponent = (function() {
    function SemanticSegmentComponent() {}
    __decorate([core_1.Input("class"), __metadata('design:type', String)], SemanticSegmentComponent.prototype, "class", void 0);
    SemanticSegmentComponent = __decorate([core_1.Component({
      changeDetection: core_1.ChangeDetectionStrategy.OnPush,
      selector: "sm-segment",
      template: "<div class=\"ui segment {{class}}\">\n  <p><ng-content></ng-content></p>\n</div>"
    }), __metadata('design:paramtypes', [])], SemanticSegmentComponent);
    return SemanticSegmentComponent;
  }());
  exports.SemanticSegmentComponent = SemanticSegmentComponent;
  return module.exports;
});

System.registerDynamic("ng-semantic/list/list", ["@angular/core"], true, function($__require, exports, module) {
  "use strict";
  ;
  var define,
      global = this,
      GLOBAL = this;
  var __decorate = (this && this.__decorate) || function(decorators, target, key, desc) {
    var c = arguments.length,
        r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
        d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
      r = Reflect.decorate(decorators, target, key, desc);
    else
      for (var i = decorators.length - 1; i >= 0; i--)
        if (d = decorators[i])
          r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
  };
  var __metadata = (this && this.__metadata) || function(k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
      return Reflect.metadata(k, v);
  };
  var core_1 = $__require('@angular/core');
  var SemanticListComponent = (function() {
    function SemanticListComponent() {}
    SemanticListComponent = __decorate([core_1.Component({
      changeDetection: core_1.ChangeDetectionStrategy.OnPush,
      encapsulation: core_1.ViewEncapsulation.None,
      selector: "sm-list",
      styles: ["sm-list { display: block; }"],
      template: "<ng-content></ng-content>"
    }), __metadata('design:paramtypes', [])], SemanticListComponent);
    return SemanticListComponent;
  }());
  exports.SemanticListComponent = SemanticListComponent;
  return module.exports;
});

System.registerDynamic("ng-semantic/button/button", ["@angular/core"], true, function($__require, exports, module) {
  "use strict";
  ;
  var define,
      global = this,
      GLOBAL = this;
  var __decorate = (this && this.__decorate) || function(decorators, target, key, desc) {
    var c = arguments.length,
        r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
        d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
      r = Reflect.decorate(decorators, target, key, desc);
    else
      for (var i = decorators.length - 1; i >= 0; i--)
        if (d = decorators[i])
          r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
  };
  var __metadata = (this && this.__metadata) || function(k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
      return Reflect.metadata(k, v);
  };
  var core_1 = $__require('@angular/core');
  var SemanticButtonComponent = (function() {
    function SemanticButtonComponent() {
      this.disabled = false;
    }
    __decorate([core_1.Input("class"), __metadata('design:type', String)], SemanticButtonComponent.prototype, "class", void 0);
    __decorate([core_1.Input("icon"), __metadata('design:type', String)], SemanticButtonComponent.prototype, "icon", void 0);
    __decorate([core_1.Input("disabled"), __metadata('design:type', Boolean)], SemanticButtonComponent.prototype, "disabled", void 0);
    SemanticButtonComponent = __decorate([core_1.Component({
      changeDetection: core_1.ChangeDetectionStrategy.OnPush,
      selector: "sm-button",
      template: "<button [ngClass]=\"{icon: icon}\" [attr.disabled]=\"disabled ? true : null\" class=\"ui {{class}} button\">\n  <i *ngIf=\"icon\" class=\"{{icon}} icon\"></i>\n    <ng-content></ng-content>\n</button>"
    }), __metadata('design:paramtypes', [])], SemanticButtonComponent);
    return SemanticButtonComponent;
  }());
  exports.SemanticButtonComponent = SemanticButtonComponent;
  return module.exports;
});

System.registerDynamic("ng-semantic/item/item", ["@angular/core"], true, function($__require, exports, module) {
  "use strict";
  ;
  var define,
      global = this,
      GLOBAL = this;
  var __decorate = (this && this.__decorate) || function(decorators, target, key, desc) {
    var c = arguments.length,
        r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
        d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
      r = Reflect.decorate(decorators, target, key, desc);
    else
      for (var i = decorators.length - 1; i >= 0; i--)
        if (d = decorators[i])
          r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
  };
  var __metadata = (this && this.__metadata) || function(k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
      return Reflect.metadata(k, v);
  };
  var core_1 = $__require('@angular/core');
  var SemanticItemComponent = (function() {
    function SemanticItemComponent() {}
    __decorate([core_1.Input("icon"), __metadata('design:type', String)], SemanticItemComponent.prototype, "icon", void 0);
    __decorate([core_1.Input("header"), __metadata('design:type', String)], SemanticItemComponent.prototype, "header", void 0);
    __decorate([core_1.Input("image"), __metadata('design:type', String)], SemanticItemComponent.prototype, "image", void 0);
    SemanticItemComponent = __decorate([core_1.Component({
      changeDetection: core_1.ChangeDetectionStrategy.OnPush,
      selector: "a[sm-item], sm-item",
      template: "<i *ngIf=\"icon\" class=\"{{icon}} icon\"></i>\n<img *ngIf=\"image\" class=\"ui avatar image\" src=\"{{image}}\">\n<div class=\"content\">\n  <div *ngIf=\"header\" class=\"header\">\n    {{header}}\n  </div>\n  <ng-content></ng-content>\n</div>"
    }), __metadata('design:paramtypes', [])], SemanticItemComponent);
    return SemanticItemComponent;
  }());
  exports.SemanticItemComponent = SemanticItemComponent;
  return module.exports;
});

System.registerDynamic("ng-semantic/dropdown/dropdown", ["@angular/core"], true, function($__require, exports, module) {
  "use strict";
  ;
  var define,
      global = this,
      GLOBAL = this;
  var __decorate = (this && this.__decorate) || function(decorators, target, key, desc) {
    var c = arguments.length,
        r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
        d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
      r = Reflect.decorate(decorators, target, key, desc);
    else
      for (var i = decorators.length - 1; i >= 0; i--)
        if (d = decorators[i])
          r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
  };
  var __metadata = (this && this.__metadata) || function(k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
      return Reflect.metadata(k, v);
  };
  var core_1 = $__require('@angular/core');
  var SemanticDropdownComponent = (function() {
    function SemanticDropdownComponent() {
      this.onChange = new core_1.EventEmitter();
    }
    SemanticDropdownComponent.prototype.ngAfterViewInit = function() {
      var _this = this;
      jQuery(this.dropdown.nativeElement).dropdown(this.options || {}).dropdown({onChange: function(value) {
          return _this.onChange.emit(value);
        }});
    };
    __decorate([core_1.Input(), __metadata('design:type', String)], SemanticDropdownComponent.prototype, "class", void 0);
    __decorate([core_1.Input(), __metadata('design:type', String)], SemanticDropdownComponent.prototype, "title", void 0);
    __decorate([core_1.Input(), __metadata('design:type', Array)], SemanticDropdownComponent.prototype, "items", void 0);
    __decorate([core_1.Input(), __metadata('design:type', Object)], SemanticDropdownComponent.prototype, "options", void 0);
    __decorate([core_1.Output(), __metadata('design:type', core_1.EventEmitter)], SemanticDropdownComponent.prototype, "onChange", void 0);
    __decorate([core_1.ViewChild("dropdown"), __metadata('design:type', core_1.ElementRef)], SemanticDropdownComponent.prototype, "dropdown", void 0);
    SemanticDropdownComponent = __decorate([core_1.Component({
      changeDetection: core_1.ChangeDetectionStrategy.OnPush,
      selector: "sm-dropdown",
      template: "\n<div class=\"ui pointing dropdown link item\" #dropdown>\n    <span class=\"text\">{{title}}</span>\n    <i class=\"dropdown icon\"></i>\n    <div class=\"menu\">\n      <div class=\"item\" *ngFor=\"let item of items\">{{item}}</div>\n    </div>\n  </div>\n"
    }), __metadata('design:paramtypes', [])], SemanticDropdownComponent);
    return SemanticDropdownComponent;
  }());
  exports.SemanticDropdownComponent = SemanticDropdownComponent;
  return module.exports;
});

System.registerDynamic("ng-semantic/select/select", ["@angular/core", "@angular/forms"], true, function($__require, exports, module) {
  "use strict";
  ;
  var define,
      global = this,
      GLOBAL = this;
  var __decorate = (this && this.__decorate) || function(decorators, target, key, desc) {
    var c = arguments.length,
        r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
        d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
      r = Reflect.decorate(decorators, target, key, desc);
    else
      for (var i = decorators.length - 1; i >= 0; i--)
        if (d = decorators[i])
          r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
  };
  var __metadata = (this && this.__metadata) || function(k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
      return Reflect.metadata(k, v);
  };
  var core_1 = $__require('@angular/core');
  var forms_1 = $__require('@angular/forms');
  var SemanticSelectComponent = (function() {
    function SemanticSelectComponent() {
      this.multiple = false;
      this.onChange = new core_1.EventEmitter();
    }
    SemanticSelectComponent.prototype.ngAfterViewInit = function() {
      var _this = this;
      if (typeof this.class === "string" && this.class.search("multiple") >= 0) {
        this.select.nativeElement.setAttribute("multiple", true);
      }
      jQuery(this.select.nativeElement).dropdown(this.options || {}).dropdown({onChange: function(value) {
          return _this.onChange.emit(value);
        }});
    };
    __decorate([core_1.Input(), __metadata('design:type', forms_1.FormControl)], SemanticSelectComponent.prototype, "control", void 0);
    __decorate([core_1.Input(), __metadata('design:type', String)], SemanticSelectComponent.prototype, "class", void 0);
    __decorate([core_1.Input(), __metadata('design:type', Object)], SemanticSelectComponent.prototype, "options", void 0);
    __decorate([core_1.Input(), __metadata('design:type', String)], SemanticSelectComponent.prototype, "placeholder", void 0);
    __decorate([core_1.Output(), __metadata('design:type', core_1.EventEmitter)], SemanticSelectComponent.prototype, "onChange", void 0);
    __decorate([core_1.ViewChild("select"), __metadata('design:type', core_1.ElementRef)], SemanticSelectComponent.prototype, "select", void 0);
    SemanticSelectComponent = __decorate([core_1.Component({
      changeDetection: core_1.ChangeDetectionStrategy.OnPush,
      directives: [forms_1.REACTIVE_FORM_DIRECTIVES],
      selector: "sm-select",
      template: "<select [formControl]=\"control\" class=\"ui {{class}} dropdown\"  #select>\n        <option value=\"\">{{placeholder}}</option>\n        <ng-content></ng-content>\n      </select>"
    }), __metadata('design:paramtypes', [])], SemanticSelectComponent);
    return SemanticSelectComponent;
  }());
  exports.SemanticSelectComponent = SemanticSelectComponent;
  return module.exports;
});

System.registerDynamic("ng-semantic/loader/loader", ["@angular/core"], true, function($__require, exports, module) {
  "use strict";
  ;
  var define,
      global = this,
      GLOBAL = this;
  var __decorate = (this && this.__decorate) || function(decorators, target, key, desc) {
    var c = arguments.length,
        r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
        d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
      r = Reflect.decorate(decorators, target, key, desc);
    else
      for (var i = decorators.length - 1; i >= 0; i--)
        if (d = decorators[i])
          r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
  };
  var __metadata = (this && this.__metadata) || function(k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
      return Reflect.metadata(k, v);
  };
  var core_1 = $__require('@angular/core');
  var SemanticLoaderComponent = (function() {
    function SemanticLoaderComponent() {
      this.complete = false;
    }
    __decorate([core_1.Input("class"), __metadata('design:type', String)], SemanticLoaderComponent.prototype, "class", void 0);
    __decorate([core_1.Input("text"), __metadata('design:type', String)], SemanticLoaderComponent.prototype, "text", void 0);
    __decorate([core_1.Input("complete"), __metadata('design:type', Boolean)], SemanticLoaderComponent.prototype, "complete", void 0);
    SemanticLoaderComponent = __decorate([core_1.Component({
      changeDetection: core_1.ChangeDetectionStrategy.OnPush,
      selector: "sm-loader",
      template: "<div *ngIf=\"!complete\" class=\"ui active dimmer {{class}}\">\n    <div [ngClass]=\"{text: text}\" class=\"ui loader\">{{text}}</div>\n  </div>"
    }), __metadata('design:paramtypes', [])], SemanticLoaderComponent);
    return SemanticLoaderComponent;
  }());
  exports.SemanticLoaderComponent = SemanticLoaderComponent;
  return module.exports;
});

System.registerDynamic("ng-semantic/card/card", ["@angular/core"], true, function($__require, exports, module) {
  "use strict";
  ;
  var define,
      global = this,
      GLOBAL = this;
  var __decorate = (this && this.__decorate) || function(decorators, target, key, desc) {
    var c = arguments.length,
        r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
        d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
      r = Reflect.decorate(decorators, target, key, desc);
    else
      for (var i = decorators.length - 1; i >= 0; i--)
        if (d = decorators[i])
          r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
  };
  var __metadata = (this && this.__metadata) || function(k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
      return Reflect.metadata(k, v);
  };
  var core_1 = $__require('@angular/core');
  var SemanticCardComponent = (function() {
    function SemanticCardComponent() {}
    __decorate([core_1.Input(), __metadata('design:type', String)], SemanticCardComponent.prototype, "class", void 0);
    __decorate([core_1.Input(), __metadata('design:type', String)], SemanticCardComponent.prototype, "image", void 0);
    SemanticCardComponent = __decorate([core_1.Component({
      changeDetection: core_1.ChangeDetectionStrategy.OnPush,
      selector: "sm-card",
      template: "<div class=\"ui {{class}}\">\n<div class=\"image\" *ngIf=\"image\">\n    <img src=\"{{image}}\">\n  </div>\n  <div class=\"content\">\n    <ng-content select=\"card-title\"></ng-content>\n    <div class=\"meta\">\n      <ng-content select=\"card-subtitle\"></ng-content>\n    </div>\n    <div class=\"description\">\n      <ng-content select=\"card-content\"></ng-content>\n    </div>\n  </div>\n  <div class=\"extra content\">\n    <ng-content select=\"card-extra\"></ng-content>\n  </div>\n  <ng-content></ng-content>\n</div>"
    }), __metadata('design:paramtypes', [])], SemanticCardComponent);
    return SemanticCardComponent;
  }());
  exports.SemanticCardComponent = SemanticCardComponent;
  var SemanticCardsComponent = (function() {
    function SemanticCardsComponent() {}
    __decorate([core_1.Input(), __metadata('design:type', String)], SemanticCardsComponent.prototype, "class", void 0);
    SemanticCardsComponent = __decorate([core_1.Component({
      changeDetection: core_1.ChangeDetectionStrategy.OnPush,
      selector: "sm-cards",
      template: "<div class=\"ui cards {{class}}\"><ng-content></ng-content></div>"
    }), __metadata('design:paramtypes', [])], SemanticCardsComponent);
    return SemanticCardsComponent;
  }());
  exports.SemanticCardsComponent = SemanticCardsComponent;
  return module.exports;
});

System.registerDynamic("ng-semantic/input/input", ["@angular/core", "@angular/forms"], true, function($__require, exports, module) {
  "use strict";
  ;
  var define,
      global = this,
      GLOBAL = this;
  var __decorate = (this && this.__decorate) || function(decorators, target, key, desc) {
    var c = arguments.length,
        r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
        d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
      r = Reflect.decorate(decorators, target, key, desc);
    else
      for (var i = decorators.length - 1; i >= 0; i--)
        if (d = decorators[i])
          r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
  };
  var __metadata = (this && this.__metadata) || function(k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
      return Reflect.metadata(k, v);
  };
  var core_1 = $__require('@angular/core');
  var forms_1 = $__require('@angular/forms');
  var SemanticInputComponent = (function() {
    function SemanticInputComponent() {}
    __decorate([core_1.Input("control"), __metadata('design:type', forms_1.FormControl)], SemanticInputComponent.prototype, "control", void 0);
    __decorate([core_1.Input("label"), __metadata('design:type', String)], SemanticInputComponent.prototype, "label", void 0);
    __decorate([core_1.Input("placeholder"), __metadata('design:type', String)], SemanticInputComponent.prototype, "placeholder", void 0);
    SemanticInputComponent = __decorate([core_1.Component({
      changeDetection: core_1.ChangeDetectionStrategy.OnPush,
      directives: [forms_1.REACTIVE_FORM_DIRECTIVES],
      selector: "sm-input",
      template: "<div class=\"field\" [ngClass]=\"{error: (!control.valid && control.dirty) }\">\n  <label *ngIf=\"label\">{{label}}</label>\n  <input type=\"text\" [formControl]=\"control\" placeholder=\"{{placeholder}}\">\n</div>"
    }), __metadata('design:paramtypes', [])], SemanticInputComponent);
    return SemanticInputComponent;
  }());
  exports.SemanticInputComponent = SemanticInputComponent;
  var SemanticCheckboxComponent = (function() {
    function SemanticCheckboxComponent() {}
    __decorate([core_1.Input("control"), __metadata('design:type', forms_1.FormControl)], SemanticCheckboxComponent.prototype, "control", void 0);
    __decorate([core_1.Input("label"), __metadata('design:type', String)], SemanticCheckboxComponent.prototype, "label", void 0);
    SemanticCheckboxComponent = __decorate([core_1.Component({
      changeDetection: core_1.ChangeDetectionStrategy.OnPush,
      directives: [forms_1.REACTIVE_FORM_DIRECTIVES],
      selector: "sm-checkbox",
      template: "<div class=\"field\" [ngClass]=\"{error: (!control.valid) }\">\n    <div class=\"ui checkbox\">\n      <input type=\"checkbox\" tabindex=\"0\" [formControl]=\"control\">\n      <label *ngIf=\"label\">{{label}}</label>\n    </div>\n  </div>"
    }), __metadata('design:paramtypes', [])], SemanticCheckboxComponent);
    return SemanticCheckboxComponent;
  }());
  exports.SemanticCheckboxComponent = SemanticCheckboxComponent;
  var SemanticTextareaComponent = (function() {
    function SemanticTextareaComponent() {}
    __decorate([core_1.Input("control"), __metadata('design:type', forms_1.FormControl)], SemanticTextareaComponent.prototype, "control", void 0);
    __decorate([core_1.Input("label"), __metadata('design:type', String)], SemanticTextareaComponent.prototype, "label", void 0);
    __decorate([core_1.Input("rows"), __metadata('design:type', String)], SemanticTextareaComponent.prototype, "rows", void 0);
    SemanticTextareaComponent = __decorate([core_1.Component({
      changeDetection: core_1.ChangeDetectionStrategy.OnPush,
      directives: [forms_1.REACTIVE_FORM_DIRECTIVES],
      selector: "sm-textarea",
      template: "<div class=\"field\" [ngClass]=\"{error: (!control.valid && control.dirty) }\">\n    <label *ngIf=\"label\">{{label}}</label>\n    <textarea rows=\"{{rows}}\" [formControl]=\"control\"></textarea>\n  </div>"
    }), __metadata('design:paramtypes', [])], SemanticTextareaComponent);
    return SemanticTextareaComponent;
  }());
  exports.SemanticTextareaComponent = SemanticTextareaComponent;
  var SemanticFormComponent = (function() {
    function SemanticFormComponent() {}
    SemanticFormComponent = __decorate([core_1.Component({
      directives: [forms_1.REACTIVE_FORM_DIRECTIVES],
      providers: [forms_1.FORM_PROVIDERS],
      selector: "form[smForm]",
      template: "<ng-content></ng-content>"
    }), __metadata('design:paramtypes', [])], SemanticFormComponent);
    return SemanticFormComponent;
  }());
  exports.SemanticFormComponent = SemanticFormComponent;
  return module.exports;
});

System.registerDynamic("ng-semantic/sidebar/sidebar", ["@angular/core"], true, function($__require, exports, module) {
  "use strict";
  ;
  var define,
      global = this,
      GLOBAL = this;
  var __decorate = (this && this.__decorate) || function(decorators, target, key, desc) {
    var c = arguments.length,
        r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
        d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
      r = Reflect.decorate(decorators, target, key, desc);
    else
      for (var i = decorators.length - 1; i >= 0; i--)
        if (d = decorators[i])
          r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
  };
  var __metadata = (this && this.__metadata) || function(k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
      return Reflect.metadata(k, v);
  };
  var core_1 = $__require('@angular/core');
  jQuery.fn.fixSidebar = function() {
    var allModules = jQuery(this);
    allModules.each(function() {
      var selector = {pusher: ".pusher"},
          module = jQuery(this),
          context = jQuery("body");
      if (module.nextAll(selector.pusher).length === 0) {
        module.detach().prependTo(context);
      }
    });
    return this;
  };
  var SemanticSidebarComponent = (function() {
    function SemanticSidebarComponent(renderer) {
      this.renderer = renderer;
      this.visible = false;
    }
    SemanticSidebarComponent.prototype.show = function(options) {
      if (!this.visible) {
        this.visible = true;
        jQuery(this.sidebar.nativeElement).sidebar(options || {}).sidebar("toggle");
      }
    };
    SemanticSidebarComponent.prototype.hide = function() {
      if (this.visible) {
        this.visible = false;
        jQuery(this.sidebar.nativeElement).sidebar("hide");
      }
    };
    SemanticSidebarComponent.prototype.ngOnInit = function() {
      jQuery(this.sidebar.nativeElement).fixSidebar();
    };
    SemanticSidebarComponent.prototype.ngOnDestroy = function() {
      this.renderer.detachView([this.sidebar.nativeElement]);
    };
    __decorate([core_1.Input(), __metadata('design:type', String)], SemanticSidebarComponent.prototype, "class", void 0);
    __decorate([core_1.ViewChild("sidebar"), __metadata('design:type', core_1.ElementRef)], SemanticSidebarComponent.prototype, "sidebar", void 0);
    SemanticSidebarComponent = __decorate([core_1.Component({
      changeDetection: core_1.ChangeDetectionStrategy.OnPush,
      selector: "sm-sidebar",
      template: "<div class=\"ui sidebar {{class}}\" #sidebar>\n<ng-content></ng-content>\n</div>"
    }), __metadata('design:paramtypes', [core_1.Renderer])], SemanticSidebarComponent);
    return SemanticSidebarComponent;
  }());
  exports.SemanticSidebarComponent = SemanticSidebarComponent;
  return module.exports;
});

System.registerDynamic("ng-semantic/tab/tab", ["@angular/core"], true, function($__require, exports, module) {
  "use strict";
  ;
  var define,
      global = this,
      GLOBAL = this;
  var __decorate = (this && this.__decorate) || function(decorators, target, key, desc) {
    var c = arguments.length,
        r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
        d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
      r = Reflect.decorate(decorators, target, key, desc);
    else
      for (var i = decorators.length - 1; i >= 0; i--)
        if (d = decorators[i])
          r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
  };
  var __metadata = (this && this.__metadata) || function(k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
      return Reflect.metadata(k, v);
  };
  var core_1 = $__require('@angular/core');
  var SemanticTabComponent = (function() {
    function SemanticTabComponent() {}
    SemanticTabComponent.prototype.ngAfterViewInit = function() {
      this.tabEl.nativeElement.parentElement.classList.add("ui", "tab", "bottom", "attached", "segment");
      if (this.active) {
        this.tabEl.nativeElement.parentElement.classList.add("active");
      }
    };
    __decorate([core_1.Input(), __metadata('design:type', Number)], SemanticTabComponent.prototype, "tab", void 0);
    __decorate([core_1.Input(), __metadata('design:type', String)], SemanticTabComponent.prototype, "title", void 0);
    __decorate([core_1.Input(), __metadata('design:type', Boolean)], SemanticTabComponent.prototype, "active", void 0);
    __decorate([core_1.ViewChild("tab"), __metadata('design:type', core_1.ElementRef)], SemanticTabComponent.prototype, "tabEl", void 0);
    SemanticTabComponent = __decorate([core_1.Component({
      selector: "sm-tab",
      template: "\n<div #tab>\n    <ng-content></ng-content>\n</div>  "
    }), __metadata('design:paramtypes', [])], SemanticTabComponent);
    return SemanticTabComponent;
  }());
  exports.SemanticTabComponent = SemanticTabComponent;
  var SemanticTabsComponent = (function() {
    function SemanticTabsComponent(elementRef) {
      this.elementRef = elementRef;
    }
    SemanticTabsComponent.prototype.ngAfterViewInit = function() {
      this.tabs.map(function(cmp, index) {
        cmp.tabEl.nativeElement.parentElement.setAttribute("data-tab", index.toString());
      });
      jQuery(this.menu.nativeElement.getElementsByClassName("item")).tab({
        childrenOnly: true,
        context: jQuery(this.elementRef.nativeElement)
      });
    };
    __decorate([core_1.ContentChildren(SemanticTabComponent), __metadata('design:type', core_1.QueryList)], SemanticTabsComponent.prototype, "tabs", void 0);
    __decorate([core_1.ViewChild("menu"), __metadata('design:type', core_1.ElementRef)], SemanticTabsComponent.prototype, "menu", void 0);
    SemanticTabsComponent = __decorate([core_1.Component({
      directives: [SemanticTabComponent],
      selector: "sm-tabs",
      template: "<div class=\"ui top attached tabular menu\" #menu>\n  <a class=\"item\" [ngClass]=\"{'active': tab.active}\" *ngFor=\"let tab of tabs; let i = index\" [attr.data-tab]=\"i\">{{tab.title}}</a>\n</div>\n<ng-content></ng-content>\n"
    }), __metadata('design:paramtypes', [core_1.ElementRef])], SemanticTabsComponent);
    return SemanticTabsComponent;
  }());
  exports.SemanticTabsComponent = SemanticTabsComponent;
  return module.exports;
});

System.registerDynamic("ng-semantic/flag/flag", ["@angular/core"], true, function($__require, exports, module) {
  "use strict";
  ;
  var define,
      global = this,
      GLOBAL = this;
  var __decorate = (this && this.__decorate) || function(decorators, target, key, desc) {
    var c = arguments.length,
        r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
        d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
      r = Reflect.decorate(decorators, target, key, desc);
    else
      for (var i = decorators.length - 1; i >= 0; i--)
        if (d = decorators[i])
          r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
  };
  var __metadata = (this && this.__metadata) || function(k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
      return Reflect.metadata(k, v);
  };
  var core_1 = $__require('@angular/core');
  var SemanticFlagComponent = (function() {
    function SemanticFlagComponent() {}
    __decorate([core_1.Input("flag"), __metadata('design:type', String)], SemanticFlagComponent.prototype, "flag", void 0);
    SemanticFlagComponent = __decorate([core_1.Component({
      changeDetection: core_1.ChangeDetectionStrategy.OnPush,
      selector: "sm-flag",
      template: "<i class=\"{{flag}} flag\"></i>"
    }), __metadata('design:paramtypes', [])], SemanticFlagComponent);
    return SemanticFlagComponent;
  }());
  exports.SemanticFlagComponent = SemanticFlagComponent;
  return module.exports;
});

System.registerDynamic("ng-semantic/contextmenu/contextmenu", ["@angular/core"], true, function($__require, exports, module) {
  "use strict";
  ;
  var define,
      global = this,
      GLOBAL = this;
  var __decorate = (this && this.__decorate) || function(decorators, target, key, desc) {
    var c = arguments.length,
        r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
        d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
      r = Reflect.decorate(decorators, target, key, desc);
    else
      for (var i = decorators.length - 1; i >= 0; i--)
        if (d = decorators[i])
          r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
  };
  var __metadata = (this && this.__metadata) || function(k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
      return Reflect.metadata(k, v);
  };
  var core_1 = $__require('@angular/core');
  var SemanticContextMenuComponent = (function() {
    function SemanticContextMenuComponent() {
      this._position = {
        x: 0,
        y: 0
      };
      this.show = false;
    }
    Object.defineProperty(SemanticContextMenuComponent.prototype, "position", {
      set: function(data) {
        if (data) {
          this._position = data;
          this.show = true;
        }
      },
      enumerable: true,
      configurable: true
    });
    SemanticContextMenuComponent.prototype.clickedOutside = function() {
      this.show = false;
    };
    __decorate([core_1.Input(), __metadata('design:type', Array)], SemanticContextMenuComponent.prototype, "items", void 0);
    __decorate([core_1.Input("position"), __metadata('design:type', Object), __metadata('design:paramtypes', [Object])], SemanticContextMenuComponent.prototype, "position", null);
    __decorate([core_1.HostListener("document:click"), __metadata('design:type', Function), __metadata('design:paramtypes', []), __metadata('design:returntype', void 0)], SemanticContextMenuComponent.prototype, "clickedOutside", null);
    SemanticContextMenuComponent = __decorate([core_1.Component({
      changeDetection: core_1.ChangeDetectionStrategy.OnPush,
      selector: "sm-contextmenu",
      styles: [".sm-contextmenu { position: fixed; z-index: 1000; }"],
      template: "<div \n[style.left.px]=\"_position?.x\" [style.top.px]=\"_position?.y\"\n[ngClass]=\"{'active visible': show}\" class=\"ui dropdown sm-contextmenu\">\n <div \n [ngClass]=\"{'visible animating slide down in': show, 'hidden': !show}\"  class=\"ui menu transition\">\n <a (click)=\"menu.action()\" *ngFor=\"let menu of items\" class=\"item\">\n <i class=\"{{menu.icon}} icon\"></i>\n {{menu.title}}\n </a>\n </div>\n </div>\n"
    }), __metadata('design:paramtypes', [])], SemanticContextMenuComponent);
    return SemanticContextMenuComponent;
  }());
  exports.SemanticContextMenuComponent = SemanticContextMenuComponent;
  return module.exports;
});

System.registerDynamic("ng-semantic/progress/progress", ["@angular/core"], true, function($__require, exports, module) {
  "use strict";
  ;
  var define,
      global = this,
      GLOBAL = this;
  var __decorate = (this && this.__decorate) || function(decorators, target, key, desc) {
    var c = arguments.length,
        r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
        d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
      r = Reflect.decorate(decorators, target, key, desc);
    else
      for (var i = decorators.length - 1; i >= 0; i--)
        if (d = decorators[i])
          r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
  };
  var __metadata = (this && this.__metadata) || function(k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
      return Reflect.metadata(k, v);
  };
  var core_1 = $__require('@angular/core');
  var SemanticProgressComponent = (function() {
    function SemanticProgressComponent() {
      this._progress = 0;
    }
    Object.defineProperty(SemanticProgressComponent.prototype, "progress", {
      set: function(value) {
        this._progress = (value >= 100) ? 100 : value;
      },
      enumerable: true,
      configurable: true
    });
    __decorate([core_1.Input("label"), __metadata('design:type', String)], SemanticProgressComponent.prototype, "label", void 0);
    __decorate([core_1.Input("class"), __metadata('design:type', String)], SemanticProgressComponent.prototype, "class", void 0);
    __decorate([core_1.Input(), __metadata('design:type', Number), __metadata('design:paramtypes', [Number])], SemanticProgressComponent.prototype, "progress", null);
    SemanticProgressComponent = __decorate([core_1.Component({
      changeDetection: core_1.ChangeDetectionStrategy.OnPush,
      selector: "sm-progress",
      template: "<div [ngClass]=\"{'success': _progress === 100}\" class=\"ui {{class}} progress\" [attr.data-percent]=\"_progress\">\n  <div class=\"bar\" [ngStyle]=\"{'width': _progress + '%'}\">\n    <div class=\"progress\">{{_progress}}%</div>\n  </div>\n  <div class=\"label\" *ngIf=\"label\">{{label}}</div>\n</div>"
    }), __metadata('design:paramtypes', [])], SemanticProgressComponent);
    return SemanticProgressComponent;
  }());
  exports.SemanticProgressComponent = SemanticProgressComponent;
  return module.exports;
});

System.registerDynamic("ng-semantic/dimmer/dimmer", ["@angular/core"], true, function($__require, exports, module) {
  "use strict";
  ;
  var define,
      global = this,
      GLOBAL = this;
  var __decorate = (this && this.__decorate) || function(decorators, target, key, desc) {
    var c = arguments.length,
        r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
        d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
      r = Reflect.decorate(decorators, target, key, desc);
    else
      for (var i = decorators.length - 1; i >= 0; i--)
        if (d = decorators[i])
          r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
  };
  var __metadata = (this && this.__metadata) || function(k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
      return Reflect.metadata(k, v);
  };
  var core_1 = $__require('@angular/core');
  var SemanticDimmerComponent = (function() {
    function SemanticDimmerComponent() {}
    SemanticDimmerComponent.prototype.show = function(options) {
      jQuery(this.dimmer.nativeElement).dimmer(options || {}).dimmer("toggle");
    };
    __decorate([core_1.ViewChild("dimmer"), __metadata('design:type', core_1.ElementRef)], SemanticDimmerComponent.prototype, "dimmer", void 0);
    SemanticDimmerComponent = __decorate([core_1.Component({
      changeDetection: core_1.ChangeDetectionStrategy.OnPush,
      selector: "sm-dimmer",
      template: "<div class=\"ui page dimmer\" #dimmer>\n  <div class=\"content\">\n    <ng-content></ng-content>\n  </div>\n</div>"
    }), __metadata('design:paramtypes', [])], SemanticDimmerComponent);
    return SemanticDimmerComponent;
  }());
  exports.SemanticDimmerComponent = SemanticDimmerComponent;
  return module.exports;
});

System.registerDynamic("ng-semantic/accordion/accordion", ["@angular/core"], true, function($__require, exports, module) {
  "use strict";
  ;
  var define,
      global = this,
      GLOBAL = this;
  var __decorate = (this && this.__decorate) || function(decorators, target, key, desc) {
    var c = arguments.length,
        r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
        d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
      r = Reflect.decorate(decorators, target, key, desc);
    else
      for (var i = decorators.length - 1; i >= 0; i--)
        if (d = decorators[i])
          r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
  };
  var __metadata = (this && this.__metadata) || function(k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
      return Reflect.metadata(k, v);
  };
  var core_1 = $__require('@angular/core');
  var SMAccordionDirective = (function() {
    function SMAccordionDirective(element) {
      this.element = element;
    }
    SMAccordionDirective.prototype.ngOnInit = function() {
      jQuery(this.element.nativeElement).accordion(this.smDirAccordion || {});
    };
    __decorate([core_1.Input(), __metadata('design:type', String)], SMAccordionDirective.prototype, "smDirAccordion", void 0);
    SMAccordionDirective = __decorate([core_1.Directive({selector: "[smDirAccordion]"}), __metadata('design:paramtypes', [core_1.ElementRef])], SMAccordionDirective);
    return SMAccordionDirective;
  }());
  var SemanticAccordionComponent = (function() {
    function SemanticAccordionComponent() {}
    __decorate([core_1.Input(), __metadata('design:type', String)], SemanticAccordionComponent.prototype, "class", void 0);
    __decorate([core_1.Input(), __metadata('design:type', String)], SemanticAccordionComponent.prototype, "options", void 0);
    SemanticAccordionComponent = __decorate([core_1.Component({
      changeDetection: core_1.ChangeDetectionStrategy.OnPush,
      directives: [SMAccordionDirective],
      selector: "sm-accordion",
      styles: ["sm-accordion sm-accordion-item:first-child .title { border-top: none !important; }"],
      template: "\n<div class=\"ui accordion {{class}}\" [smDirAccordion]=\"options\">\n    <ng-content></ng-content>\n</div>\n"
    }), __metadata('design:paramtypes', [])], SemanticAccordionComponent);
    return SemanticAccordionComponent;
  }());
  exports.SemanticAccordionComponent = SemanticAccordionComponent;
  var SemanticAccordionItemComponent = (function() {
    function SemanticAccordionItemComponent() {}
    __decorate([core_1.Input(), __metadata('design:type', String)], SemanticAccordionItemComponent.prototype, "class", void 0);
    SemanticAccordionItemComponent = __decorate([core_1.Component({
      changeDetection: core_1.ChangeDetectionStrategy.OnPush,
      selector: "sm-accordion-item",
      template: "\n<div class=\"{{class}} title\">\n    <i class=\"dropdown icon\"></i>\n    <ng-content select=\"accordion-title\"></ng-content>\n</div>\n<div class=\"{{class}} content\">\n    <ng-content select=\"accordion-content\"></ng-content>\n</div>\n"
    }), __metadata('design:paramtypes', [])], SemanticAccordionItemComponent);
    return SemanticAccordionItemComponent;
  }());
  exports.SemanticAccordionItemComponent = SemanticAccordionItemComponent;
  return module.exports;
});

System.registerDynamic("ng-semantic/rating/rating", ["@angular/core"], true, function($__require, exports, module) {
  "use strict";
  ;
  var define,
      global = this,
      GLOBAL = this;
  var __decorate = (this && this.__decorate) || function(decorators, target, key, desc) {
    var c = arguments.length,
        r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
        d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
      r = Reflect.decorate(decorators, target, key, desc);
    else
      for (var i = decorators.length - 1; i >= 0; i--)
        if (d = decorators[i])
          r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
  };
  var __metadata = (this && this.__metadata) || function(k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
      return Reflect.metadata(k, v);
  };
  var core_1 = $__require('@angular/core');
  var SemanticRatingComponent = (function() {
    function SemanticRatingComponent() {
      this.onRate = new core_1.EventEmitter();
    }
    SemanticRatingComponent.prototype.ngAfterViewInit = function() {
      var _this = this;
      jQuery(this.rating.nativeElement).rating({
        initialRating: this.initialRating || 0,
        maxRating: this.maxRating || 5,
        onRate: function(value) {
          _this.onRate.emit(value);
        }
      });
    };
    __decorate([core_1.Input(), __metadata('design:type', String)], SemanticRatingComponent.prototype, "class", void 0);
    __decorate([core_1.Input(), __metadata('design:type', Number)], SemanticRatingComponent.prototype, "initialRating", void 0);
    __decorate([core_1.Input(), __metadata('design:type', Number)], SemanticRatingComponent.prototype, "maxRating", void 0);
    __decorate([core_1.Output(), __metadata('design:type', core_1.EventEmitter)], SemanticRatingComponent.prototype, "onRate", void 0);
    __decorate([core_1.ViewChild("rating"), __metadata('design:type', core_1.ElementRef)], SemanticRatingComponent.prototype, "rating", void 0);
    SemanticRatingComponent = __decorate([core_1.Component({
      changeDetection: core_1.ChangeDetectionStrategy.OnPush,
      selector: "sm-rating",
      template: "<div class=\"ui {{class}} rating\" #rating></div>"
    }), __metadata('design:paramtypes', [])], SemanticRatingComponent);
    return SemanticRatingComponent;
  }());
  exports.SemanticRatingComponent = SemanticRatingComponent;
  return module.exports;
});

System.registerDynamic("ng-semantic", ["./ng-semantic/menu/menu", "./ng-semantic/message/message", "./ng-semantic/popup/tooltip", "./ng-semantic/popup/popup", "./ng-semantic/modal/modal", "./ng-semantic/segment/segment", "./ng-semantic/list/list", "./ng-semantic/button/button", "./ng-semantic/item/item", "./ng-semantic/dropdown/dropdown", "./ng-semantic/select/select", "./ng-semantic/loader/loader", "./ng-semantic/card/card", "./ng-semantic/input/input", "./ng-semantic/sidebar/sidebar", "./ng-semantic/tab/tab", "./ng-semantic/flag/flag", "./ng-semantic/contextmenu/contextmenu", "./ng-semantic/progress/progress", "./ng-semantic/dimmer/dimmer", "./ng-semantic/accordion/accordion", "./ng-semantic/rating/rating"], true, function($__require, exports, module) {
  "use strict";
  ;
  var define,
      global = this,
      GLOBAL = this;
  function __export(m) {
    for (var p in m)
      if (!exports.hasOwnProperty(p))
        exports[p] = m[p];
  }
  var menu_1 = $__require('./ng-semantic/menu/menu');
  var message_1 = $__require('./ng-semantic/message/message');
  var tooltip_1 = $__require('./ng-semantic/popup/tooltip');
  var popup_1 = $__require('./ng-semantic/popup/popup');
  var modal_1 = $__require('./ng-semantic/modal/modal');
  var segment_1 = $__require('./ng-semantic/segment/segment');
  var list_1 = $__require('./ng-semantic/list/list');
  var button_1 = $__require('./ng-semantic/button/button');
  var item_1 = $__require('./ng-semantic/item/item');
  var dropdown_1 = $__require('./ng-semantic/dropdown/dropdown');
  var select_1 = $__require('./ng-semantic/select/select');
  var loader_1 = $__require('./ng-semantic/loader/loader');
  var card_1 = $__require('./ng-semantic/card/card');
  var input_1 = $__require('./ng-semantic/input/input');
  var sidebar_1 = $__require('./ng-semantic/sidebar/sidebar');
  var tab_1 = $__require('./ng-semantic/tab/tab');
  var flag_1 = $__require('./ng-semantic/flag/flag');
  var contextmenu_1 = $__require('./ng-semantic/contextmenu/contextmenu');
  var progress_1 = $__require('./ng-semantic/progress/progress');
  var dimmer_1 = $__require('./ng-semantic/dimmer/dimmer');
  var accordion_1 = $__require('./ng-semantic/accordion/accordion');
  var rating_1 = $__require('./ng-semantic/rating/rating');
  __export($__require('./ng-semantic/menu/menu'));
  __export($__require('./ng-semantic/contextmenu/contextmenu'));
  __export($__require('./ng-semantic/message/message'));
  __export($__require('./ng-semantic/popup/tooltip'));
  __export($__require('./ng-semantic/popup/popup'));
  __export($__require('./ng-semantic/modal/modal'));
  __export($__require('./ng-semantic/segment/segment'));
  __export($__require('./ng-semantic/rating/rating'));
  __export($__require('./ng-semantic/list/list'));
  __export($__require('./ng-semantic/button/button'));
  __export($__require('./ng-semantic/item/item'));
  __export($__require('./ng-semantic/dropdown/dropdown'));
  __export($__require('./ng-semantic/select/select'));
  __export($__require('./ng-semantic/loader/loader'));
  __export($__require('./ng-semantic/card/card'));
  __export($__require('./ng-semantic/input/input'));
  __export($__require('./ng-semantic/sidebar/sidebar'));
  __export($__require('./ng-semantic/tab/tab'));
  __export($__require('./ng-semantic/flag/flag'));
  __export($__require('./ng-semantic/progress/progress'));
  __export($__require('./ng-semantic/dimmer/dimmer'));
  __export($__require('./ng-semantic/accordion/accordion'));
  exports.SEMANTIC_COMPONENTS = [card_1.SemanticCardComponent, card_1.SemanticCardsComponent, contextmenu_1.SemanticContextMenuComponent, input_1.SemanticInputComponent, input_1.SemanticTextareaComponent, input_1.SemanticCheckboxComponent, input_1.SemanticFormComponent, menu_1.SemanticMenuComponent, message_1.SemanticMessageComponent, segment_1.SemanticSegmentComponent, dimmer_1.SemanticDimmerComponent, popup_1.SemanticPopupComponent, dropdown_1.SemanticDropdownComponent, list_1.SemanticListComponent, select_1.SemanticSelectComponent, flag_1.SemanticFlagComponent, item_1.SemanticItemComponent, sidebar_1.SemanticSidebarComponent, progress_1.SemanticProgressComponent, modal_1.SemanticModalComponent, tab_1.SemanticTabsComponent, tab_1.SemanticTabComponent, button_1.SemanticButtonComponent, loader_1.SemanticLoaderComponent, accordion_1.SemanticAccordionComponent, accordion_1.SemanticAccordionItemComponent, rating_1.SemanticRatingComponent];
  exports.SEMANTIC_DIRECTIVES = [tooltip_1.SMTooltipDirective];
  return module.exports;
});
